prow/plugins/cherrypickapproved/cherrypickapproved.go

```diff
@@ -0,0 +1,228 @@
+*/
+
+package cherrypickapproved
+
+import (
+	"errors"
+	"fmt"
+	"io"
+	"net/http"
+	"regexp"
+	"strings"
+	"time"
+
+	"github.com/sirupsen/logrus"
+	"k8s.io/test-infra/prow/config"
+	"k8s.io/test-infra/prow/github"
+	"k8s.io/test-infra/prow/labels"
+	"k8s.io/test-infra/prow/pluginhelp"
+	"k8s.io/test-infra/prow/plugins"
+	"sigs.k8s.io/yaml"
+)
+
+const (
+	// PluginName defines this plugin's registered name.
+	PluginName = labels.CpApproved
+
+	defaultKubernetesRepoAndOrg = "kubernetes"
+)
+
+var (
+	branchRegexp = regexp.MustCompile("^release-*")
+)
+
+func init() {
+	plugins.RegisterReviewEventHandler(PluginName, handlePullRequestReviewEvent, helpProvider)
+}
+
+func helpProvider(*plugins.Configuration, []config.OrgRepo) (*pluginhelp.PluginHelp, error) {
+	return &pluginhelp.PluginHelp{
+		Description: fmt.Sprintf(
+			"The %s plugin helps Kubernetes Release Managers to approve cherry-picks by using GitHub reviews",
+			PluginName,
+		),
+	}, nil
+}
+
+type handler struct {
+	impl
+}
+
+func newHandler() *handler {
+	return &handler{
+		impl: &defaultImpl{},
+	}
+}
+
+//go:generate go run github.com/maxbrunsfeld/counterfeiter/v6 -generate
+//counterfeiter:generate . impl
+type impl interface {
+	GetIssueLabels(gc plugins.PluginGitHubClient, org, repo string, number int) ([]github.Label, error)
+	AddLabel(gc plugins.PluginGitHubClient, org, repo string, number int, label string) error
+	RemoveLabel(gc plugins.PluginGitHubClient, org, repo string, number int, label string) error
+	HttpGet(url string) (*http.Response, error)
+	CloseBody(resp *http.Response) error
+	ReadAllBody(resp *http.Response) ([]byte, error)
+	YAMLUnmarshal(y []byte) (map[string]map[string][]string, error)
+}
+
+type defaultImpl struct{}
+
+func (*defaultImpl) GetIssueLabels(gc plugins.PluginGitHubClient, org, repo string, number int) ([]github.Label, error) {
+	return gc.GetIssueLabels(org, repo, number)
+}
+
+func (*defaultImpl) AddLabel(gc plugins.PluginGitHubClient, org, repo string, number int, label string) error {
+	return gc.AddLabel(org, repo, number, label)
+
+}
+
+func (*defaultImpl) RemoveLabel(gc plugins.PluginGitHubClient, org, repo string, number int, label string) error {
+	return gc.RemoveLabel(org, repo, number, label)
+}
+
+func (*defaultImpl) HttpGet(url string) (*http.Response, error) {
+	return http.Get(url)
+}
+
+func (*defaultImpl) CloseBody(resp *http.Response) error {
+	return resp.Body.Close()
+}
+
+func (*defaultImpl) ReadAllBody(resp *http.Response) ([]byte, error) {
+	return io.ReadAll(resp.Body)
+}
+
+func (*defaultImpl) YAMLUnmarshal(y []byte) (map[string]map[string][]string, error) {
+	var result map[string]map[string][]string
+	if err := yaml.Unmarshal(y, &result); err != nil {
+		return nil, err
+	}
+	return result, nil
+}
+
+func handlePullRequestReviewEvent(pc plugins.Agent, e github.ReviewEvent) error {
+	if err := newHandler().handle(pc.Logger, pc.GitHubClient, e); err != nil {
+		pc.Logger.WithError(err).Error("skipping")
```

Comment:
I think that we should return an error here instead of just logging it.

Reply:
That's tricky since I'm not exactly sure how prow behaves here. Anyways, changing to return the `err`.